Resources/public/js/views/ez-languageselectionboxview.js

```diff
@@ -0,0 +1,402 @@
+                // TODO: existingTranslations and canBaseTranslation have influence on view
+                //          so it needs to be rendered on change of these attributes
+                // this.render();
+            });
+            this.on('selectedNewTranslationChange', function (e) {
+                if ( this.get('active') ) {
+                    this._uiHighlightTranslation('new', e.newVal);
+                }
+            });
+            this.on('selectedBaseTranslationChange', function (e) {
+                if ( this.get('active') ) {
+                    this._uiHighlightTranslation('base', e.newVal);
+                }
+            });
+            this.on('baseTranslationChange', function (e) {
+                if ( this.get('active') ) {
+                    this._uiBaseTranslationState(e.newVal);
+                }
+            });
+        },
+
+        /**
+         * Updates the title in the rendered selectionLanguageBox
+         * 
+         * @method _uiUpdateTitle
+         * @protected
+         */
+        _uiUpdateTitle: function () {
+            this.get('container')
+                .one('.ez-languageselectionbox-title').setContent(this.get('title'));
+        },
+
+        /**
+         * Publishes the cancelDiscover and contentDiscovered events
+         *
+         * @method _publishEvents
+         * @protected
+         */
+        _publishEvents: function () {
+            this.publish(CONFIRM, {
+                bubbles: true,
+                emitFacade: true,
+                preventable: false,
+                defaultFn: this._confirmLanguageSelection,
+            });
+            this.publish(CANCEL, {
+                bubbles: true,
+                emitFacade: true,
+                preventable: false,
+                defaultFn: this._cancelLanguageSelection,
+            });
+        },
+
+        /**
+         * Updates the event handler for the given event
+         *
+         * @method _syncEventHandler
+         * @protected
+         * @param {String} eventName
+         * @param {Function} oldHandler
+         * @param {Function} newHandler
+         */
+        _syncEventHandler: function (eventName, oldHandler, newHandler) {
+            this.detach(eventName);
+            this.after(eventName, function (e) {
+                if ( newHandler ) {
+                    newHandler(e);
+                }
+                this._resetState();
+            });
+        },
+
+        /**
+         * Resets the state of the languageSelectionBox view
+         *
+         * @method _resetState
+         * @protected
+         */
+        _resetState: function () {
+            this.reset('title');
+            this.reset('confirmHandler');
+            this.reset('cancelHandler');
+            this.reset('baseTranslation');
+            this.reset('selectedNewTranslation');
+            this.reset('selectedBaseTranslation');
+            this.render();
```

Comment:
I have removed it frome here. I have also removed `this.reset()` for  `baseTranslation`, `selectedNewTranslation` and `selectedBaseTranslation` as it shouldn't be done here probably.